wrappers/src/marshalling.hpp

```diff
@@ -582,10 +628,18 @@ inline realm_value_t to_capi(const object_store::Dictionary& dictionary, const M
         if ((dictionary.get_type() & ~PropertyType::Flags) == PropertyType::Object) {
             return to_capi(ObjLink(dictionary.get_object_schema().table_key, val.get<ObjKey>()), dictionary.get_realm());
         }
-
         REALM_UNREACHABLE();
     case type_TypedLink:
         return to_capi(val.get_link(), dictionary.get_realm());
+    case type_List:
+        return to_capi(new List(dictionary.get_list(key)));
```

Comment:
It is, but there are no methods like `Mixed.get_list` or similar that we can use on the `val` that we just got. The only thing we can do with that is to check the type that is contained inside, and then use the specific collection methods like `list.get_list`. 
This is what is shown in the technical design and I've just had a confirmation from Nicola, that is the way that it works now, as the `Mixed` value doesn't really contain anything inside.

Reply:
Blah. This sounds like a deficiency in the Core API design and we should file a ticket to have them address it. We should probably add a comment somewhere to clarify this is the case, otherwise, when just reading the code, it looks kind of weird.